Prompt Engineering

Asr161 opened the pull request:
Created a guide on prompt engineering and how one can structure prompts effectively. Introduced what prompt engineering is, what prompts are then highlighted the key 10 points that developers must keep in mind while structuring prompts for their particular use cases. Provided an example of a use case along with a good and a bad prompt to help the reader understand the difference. Additional resources on learning how to structure prompts effectively as also mentioned. A resource is also provide for structuring prompts to make use of LLAMA2, as an example.

nathan-cai commented on Topics/Development_Process/Prompt_Engineering.md:
```diff
@@ -0,0 +1,77 @@
+## Introduction to Prompt Engineering with LLAMA2 on VertexAI
+
+Prompt engineering is an art and science of designing input queries (prompts) to generate the desired output from language models like LLAMA2 on VertexAI. The quality of the output is directly influenced by how well the prompt is structured, making this skill vital for those utilizing language models in their applications.
```
"Prompt engineering is **the** art and science of"

Also since you mention VertexAI perhaps you should talk about PALM2 instead of LLAMA2

Asr161 replied:
Thanks for pointing that out, I have made that change.

nathan-cai commented on Topics/Development_Process/Prompt_Engineering.md:
```diff
@@ -0,0 +1,77 @@
+
+"Context: You are a teacher who wants to create multiple-choice questions based on readings given to the students.
+ Using the following information and nothing else, generate 5 Multiple Choice questions based on the information above and return them in JSON format.
+ Exactly follow the JSON format of the example question below. Each question should be separated by "&&" and nothing else.
```
This part says to follow the JSON format of the example question below, but it did not give an example. Perhaps remove this sentence?

Asr161 replied:
Good catch, I have removed that sentence which still makes that prompt a bad one. Thank you!

nathan-cai reviewed (changes requested):
Overall, the structure and content of this page is very clear and informative. I love that you provided examples of both good and bad prompts, the contrast was very helpful. I left a couple comments on small changes. Also perhaps you could add a section on choosing different LLM's and the differences that could arise. You could add more information about Llama2 and PALM2 and even GPT-4.

Asr161 commented:
I have changed the focus of the guide to be more general, I have made the few changes you requested for and added an additional section which would help users choose an LLM for their particular use case.

nathan-cai commented on Topics/Development_Process/Prompt_Engineering.md:
```diff
@@ -68,6 +69,25 @@ The provided prompt is not well-structured for the following reasons:
 3. agueness in Instruction: The second prompt is vague in its instructions, missing out on details such as the level of difficulty of the questions or the specific areas of the text to focus on, which could lead to a mismatch in expectations and actual output.
 
 
+## What LLM shoudl you choose?
```
Small spelling error here

Asr161 replied:
thank you for pointing that out, I have made the change

nathan-cai commented on Topics/Development_Process/Prompt_Engineering.md:
```diff
@@ -68,6 +69,25 @@ The provided prompt is not well-structured for the following reasons:
 3. agueness in Instruction: The second prompt is vague in its instructions, missing out on details such as the level of difficulty of the questions or the specific areas of the text to focus on, which could lead to a mismatch in expectations and actual output.
```
Small Spelling Error here

Asr161 replied:
Good catch, Have made the change

nathan-cai commented on Topics/Development_Process/Prompt_Engineering.md:
```diff
@@ -0,0 +1,97 @@
+
+2. Master Prompt Engineering: Demystifying Prompting Through a Structured Approach : (https://promptengineering.org/master-prompt-engineering-demystifying-prompting-through-a-structured-approach/#:~:text=,can%20quickly%20grasp%20your%20intent) 
+
+3. A guide to prompting llama2 : (https://replicate.com/blog/how-to-prompt-llama) 
```
This text is not formatted the same as the ones above. Also the capitalization of LlaMA2

nathan-cai commented:
Another thing to add, not to certain if this is needed but add a link to this page on the Development_Process.md Page so others can access your page.

Asr161 commented:
Thank you for you're feedback! Ive corrected the grammatical errors and added a few more resources to help readers get acquainted to different LLM's. Please do let me know if any more changes are required. Thanks!

Asr161 commented on Topics/Development_Process/Prompt_Engineering.md:
```diff
@@ -94,4 +94,18 @@ Here are few popular LLM's you can make use of:
\ No newline at end of file
+3. A guide to prompting LlaMA2 : (https://replicate.com/blog/how-to-prompt-llama) 
+
+Here are resources for the LLM's discusses above:
```
Made the change, thank you for pointing that out!

Asr161 commented:
Made the change and I've also added a link to this page on the development_process.md page for others to see.

nathan-cai reviewed (approved):
Everything Looks Good!
